mobile/library/common/network/android.cc

```diff
@@ -29,7 +29,7 @@ namespace {
 #endif
 
 void setAlternateGetifaddrs() {
-#if defined(INCLUDE_IFADDRS)
+#if defined(__ANDROID_API__)
   auto& os_syscalls = Api::OsSysCallsSingleton::get();
   ENVOY_BUG(!os_syscalls.supportsGetifaddrs(),
```

Comment:
Do we also want the min version or to change the bug call here?

Reply:
Hm. Which min version?

Reply:
24?  
https://github.com/envoyproxy/envoy/blob/main/source/common/api/posix/os_sys_calls_impl.cc#L352
/wait

Reply:
I just want to point out that this is a compilation (and not a runtime check) and that we effectively always use a custom (or 'alternate' as in the name of the method) implementation of `getifaddrs` as Android is compiled against `ndk_api_level` equal to 21. https://github.com/envoyproxy/envoy/blob/c394382139010828c125a23eccb47a5172e2eab8/mobile/WORKSPACE#L96

And yea, I think that @alyssawilk has a good point when it comes to the min version check - we probably do not want to use the custom implementation if the system one is available.

Reply:
Ah, I see. Ok, I've moved the min version check here.

That being said, I wonder if it would actually make more sense to move this Envoy Mobile alternate implementation out of Envoy Mobile and into Envoy itself in OsSysCallsImpl::supportsGetifaddrs(). This would resolve the TODO there:

 ```
 // TODO: eliminate this branching by upstreaming an alternative Android implementation
// e.g.: https://github.com/envoyproxy/envoy-mobile/blob/main/third_party/android/ifaddrs-android.h
```

Would that be a good idea?

Reply:
I took a look at this and I think this is the "right" solution. Here's PR which does the needful. What do y'all think?

https://github.com/envoyproxy/envoy/pull/24983

Reply:
I approved the other PR - let us know what you what to do with this one given that there is an overlap between it and https://github.com/envoyproxy/envoy/pull/24983